test/python/transpiler/test_high_level_synthesis.py

```diff
@@ -501,6 +503,188 @@ def test_qubits_get_passed_to_plugins(self):
+        qc = QuantumCircuit(10)
+        qc.append(perm, [3, 2, 7, 8])
+
+        coupling_map = CouplingMap.from_ring(10)
+
+        synthesis_config = HLSConfig(permutation=[("token_swapper", {"trials": 10})])
+        qc_transpiled = PassManager(
+            HighLevelSynthesis(
+                synthesis_config, coupling_map=coupling_map, target=None, use_qubit_indices=True
+            )
+        ).run(qc)
+
+        # The synthesis is not possible. In this case the plugin should return `None`
+        # and `HighLevelSynthesis` should not change the original circuit.
+        self.assertEqual(qc_transpiled, qc)
+
+    def test_abstract_synthesis_all_permutations(self):
+        """Test abstract synthesis of permutation gates, varying permutation gate patterns."""
+
+        edges = [(0, 1), (1, 0), (1, 2), (2, 1), (1, 3), (3, 1), (3, 4), (4, 3)]
+
+        coupling_map = CouplingMap()
+        for i in range(5):
+            coupling_map.add_physical_qubit(i)
+        for edge in edges:
+            coupling_map.add_edge(*edge)
+
+        synthesis_config = HLSConfig(permutation=[("token_swapper", {"trials": 10})])
+        pm = PassManager(
+            HighLevelSynthesis(
+                synthesis_config, coupling_map=coupling_map, target=None, use_qubit_indices=False
+            )
+        )
+
+        for pattern in itertools.permutations(range(4)):
+            qc = QuantumCircuit(5)
+            qc.append(PermutationGate(pattern), [2, 0, 3, 1])
+            self.assertIn("permutation", qc.count_ops())
+
+            qc_transpiled = pm.run(qc)
+            self.assertNotIn("permutation", qc_transpiled.count_ops())
+
+            self.assertEqual(Operator(qc), Operator(qc_transpiled))
+
+    def test_concrete_synthesis_all_permutations(self):
+        """Test concrete synthesis of permutation gates, varying permutation gate patterns."""
+
+        edges = [(0, 1), (1, 0), (1, 2), (2, 1), (1, 3), (3, 1), (3, 4), (4, 3)]
+
+        coupling_map = CouplingMap()
+        for i in range(5):
+            coupling_map.add_physical_qubit(i)
+        for edge in edges:
+            coupling_map.add_edge(*edge)
+
+        synthesis_config = HLSConfig(permutation=[("token_swapper", {"trials": 10})])
+        pm = PassManager(
+            HighLevelSynthesis(
+                synthesis_config, coupling_map=coupling_map, target=None, use_qubit_indices=True
+            )
+        )
+
+        for pattern in itertools.permutations(range(4)):
+
+            qc = QuantumCircuit(5)
+            qc.append(PermutationGate(pattern), [2, 0, 3, 1])
+            self.assertIn("permutation", qc.count_ops())
+
+            qc_transpiled = pm.run(qc)
+            self.assertNotIn("permutation", qc_transpiled.count_ops())
+            self.assertEqual(Operator(qc), Operator(qc_transpiled))
+
+            for inst in qc_transpiled:
+                qubits = tuple(q._index for q in inst.qubits)
+                self.assertIn(qubits, edges)
```

Comment:
I had to fix this since there is no `Qubit.index` method anymore; is using `q._index` the recommended approach?

Reply:
The best way to do this is use `circuit.find_bit(qubit).index` to find the position in a circuit.